src/main/java/com/pkb/unit/tracker/Tracker.java

```diff
@@ -26,6 +32,51 @@
 public class Tracker {
     private Tracker() {}
 
+    private static class RestartTracker implements ObservableSource<Boolean> {
```

Comment:
I'm interested to know if there's a better way of doing this, I looked at the `reduce` operation, but this only applies when the observable terminates http://reactivex.io/documentation/operators/reduce.html
I looked at the `scan` operation, but this will emit whenever an event is received. 

I _think_ I want this: http://reactivex.io/documentation/operators/and-then-when.html but it isn't available in RxJava2

Reply:
Do you want to save the "subscription to the Bus and then filter the emissions and pass the relevant one to the observer" process?

Reply:
I see your point so you want be able to implement the filtering logic in an RX operator.